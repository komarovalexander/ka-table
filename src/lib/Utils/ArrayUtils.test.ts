import {
  getCopyOfArrayAndAddItem, getCopyOfArrayAndDeleteItem, getCopyOfArrayAndInsertOrReplaceItem,
} from './ArrayUtils';

describe('ArrayUtils', () => {
  it('getCopyOfArrayAndAddItem add item to array', () => {
    const array = [1, 2];
    const result = getCopyOfArrayAndAddItem(3, array);

    expect(result).not.toEqual(array);
    expect(result).toStrictEqual([1, 2, 3]);
  });

  it('getCopyOfArrayAndAddItem add item to empty array', () => {
    const result = getCopyOfArrayAndAddItem(1, undefined);
    expect(result).toStrictEqual([1]);
  });

  it('getCopyOfArrayAndInsertOrReplaceItem replace', () => {
    const array = [
      { column1: 1,  column2: 31 },
      { column1: 2,  column2: 32 },
      { column1: 3,  column2: 33 },
      { column1: 4,  column2: 34 },
    ];
    const result = getCopyOfArrayAndInsertOrReplaceItem({ column1: 3,  column2: 333 }, 'column1', array);

    expect(result).not.toEqual(array);
    expect(result).toMatchSnapshot();
  });

  it('getCopyOfArrayAndInsertOrReplaceItem insert', () => {
    const array = [
      { column1: 1,  column2: 31 },
      { column1: 2,  column2: 32 },
      { column1: 4,  column2: 34 },
    ];
    const result = getCopyOfArrayAndInsertOrReplaceItem({ column1: 3,  column2: 333 }, 'column1', array);

    expect(result).not.toEqual(array);
    expect(result).toMatchSnapshot();
  });

  it('getCopyOfArrayAndDeleteItem', () => {
    const array = [
      { column1: 1,  column2: 31 },
      { column1: 2,  column2: 32 },
      { column1: 3,  column2: 33 },
      { column1: 4,  column2: 34 },
    ];
    const result = getCopyOfArrayAndDeleteItem({ column1: 3,  column2: 333 }, 'column1', array);

    expect(result).not.toEqual(array);
    expect(result).toMatchSnapshot();
  });
});
