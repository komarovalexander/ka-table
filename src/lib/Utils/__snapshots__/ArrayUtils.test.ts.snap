// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrayUtils getCopyOfArrayAndDeleteItem 1`] = `
Array [
  Object {
    "column1": 1,
    "column2": 31,
  },
  Object {
    "column1": 2,
    "column2": 32,
  },
  Object {
    "column1": 4,
    "column2": 34,
  },
]
`;

exports[`ArrayUtils getCopyOfArrayAndInsertOrReplaceItem insert 1`] = `
Array [
  Object {
    "column1": 1,
    "column2": 31,
  },
  Object {
    "column1": 2,
    "column2": 32,
  },
  Object {
    "column1": 4,
    "column2": 34,
  },
  Object {
    "column1": 3,
    "column2": 333,
  },
]
`;

exports[`ArrayUtils getCopyOfArrayAndInsertOrReplaceItem replace 1`] = `
Array [
  Object {
    "column1": 1,
    "column2": 31,
  },
  Object {
    "column1": 2,
    "column2": 32,
  },
  Object {
    "column1": 3,
    "column2": 333,
  },
  Object {
    "column1": 4,
    "column2": 34,
  },
]
`;
